diff --git a/capacitor/src/main/java/com/getcapacitor/plugin/util/AssetUtil.java b/capacitor/src/main/java/com/getcapacitor/plugin/util/AssetUtil.java
index 3a7043bb5b517e373ddb28a425dde08bf289d339..c2c4da82c8b7949f77a93dff4ccf133a6aa32b48 100644
--- a/capacitor/src/main/java/com/getcapacitor/plugin/util/AssetUtil.java
+++ b/capacitor/src/main/java/com/getcapacitor/plugin/util/AssetUtil.java
@@ -101,7 +101,7 @@ public final class AssetUtil {
      * @return URI pointing to the given path.
      */
     private Uri getUriFromAsset(String path) {
-        String resPath = path.replaceFirst("file:/", "www").replaceFirst("\\?.*$", "");
+        String resPath = path.replaceFirst("file:/", "public").replaceFirst("\\?.*$", "");
         String fileName = resPath.substring(resPath.lastIndexOf('/') + 1);
         File file = getTmpFile(fileName);
 
@@ -323,7 +323,7 @@ public final class AssetUtil {
      */
     private Uri getUriFromFile(File file) {
         try {
-            String authority = context.getPackageName() + ".provider";
+            String authority = context.getPackageName() + ".fileprovider";
             return FileProvider.getUriForFile(context, authority, file);
         } catch (IllegalArgumentException e) {
             Logger.error("File not supported by provider", e);
